### PR TITLE
CLN: remove unneeded try/except in nanops

### PR DESCRIPTION
**Pull request by topper-123**
Look like this try/except isn't needed, checking in CI if it verifies the result.

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -1075,11 +1075,8 @@ def reduction(
 
         if (axis is not None and values.shape[axis] == 0) or values.size == 0:
             dtype_max = _get_dtype_max(dtype)
-            try:
-                result = getattr(values, meth)(axis, dtype=dtype_max)
-                result.fill(np.nan)
-            except (AttributeError, TypeError, ValueError):
-                result = np.nan
+            result = getattr(values, meth)(axis, dtype=dtype_max)
+            result.fill(np.nan)
```
Comment (jbrockmendel) [inline pandas/core/nanops.py:1079]: do we have tests that get here?

**Comment by topper-123** [comment]
I rearranged it again. `if (axis is not None and values.shape[axis] == 0) or values.size == 0:` is always the same as `if values.size == 0:` AFAIKT.

So I've just added a `if values.size == 0` clause at the top and cleaned up even further, so it's quite simple now.

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -1067,22 +1067,14 @@ def reduction(
         axis: AxisInt | None = None,
         skipna: bool = True,
         mask: npt.NDArray[np.bool_] | None = None,
-    ) -> Dtype:
```
Comment (mroeschke) [inline pandas/core/nanops.py:1070]: Can we keep this type annotation?

File: pandas/core/nanops.py [hunk 3]
```diff
@@ -1067,22 +1067,14 @@ def reduction(
...
-        dtype = values.dtype
+    ):
+        if values.size == 0:
+            return _na_for_min_count(values, axis)
```
Comment (mroeschke) [inline pandas/core/nanops.py:1072]: I assume we have tests that get here?

File: pandas/core/nanops.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/nanops.py:1070]: This annotation was wrong and this doesn't return a `Dtype` but the the result of a min/max of a DataFrame/Series, i.e. typically a scalar or a ndarray (but the min/max of a Series can in principle be anything....)

IDK if we return type `Scalar` as annotation for Series?

File: pandas/core/nanops.py (hunk 3 above)
Comment (topper-123) [reply pandas/core/nanops.py:1072]: The `values.size == 0` case is actually never reached in the current code:

You can see this function is wrapped in `bottleneck_switch` and if you go to `bottleneck_switch` you can see (line 125) it takes care of the `values.size == 0` case, also when `use_bottleneck = False` (`kwds.get("min_count") is None` is always true for `_nanminmax`).

I just think this is more readable than the old code.

Comment (mroeschke) [reply pandas/core/nanops.py:1072]: Gotcha. Thanks for the explanation

File: pandas/core/nanops.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/nanops.py:1070]: Yeah not either, but since this is an inner function the typing here probably doesn't matter as much

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @topper-123

**Pull request merged**